nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/ExperimentsManager.cs

```diff
@@ -0,0 +1,41 @@
+using NuGetUpdater.Core.Run;
+
+namespace NuGetUpdater.Core;
+
+public record ExperimentsManager
+{
+    public bool UseLegacyDependencySolver { get; init; } = false;
+
+    public static ExperimentsManager GetExperimentsManager(Dictionary<string, object>? experiments)
+    {
+        return new ExperimentsManager()
+        {
+            UseLegacyDependencySolver = IsEnabled(experiments, "nuget_legacy_dependency_solver"),
```

Comment:
This (and the property definition 6 lines up) will now be the only changes necessary when a new experiment is added or an old one removed.